pkg/providers/instancetype/suite_test.go

```diff
@@ -1020,6 +1014,26 @@ var _ = Describe("InstanceType Provider", func() {
 					Expect(lo.FromPtr(nicDeleteOption)).To(Equal(armcompute.DeleteOptionsDelete))
 				}
 			})
+			It("should not create unneeded secondary ips for azure cni with overlay", func() {
+				ExpectApplied(ctx, env.Client, nodePool, nodeClass)
+				pod := coretest.UnschedulablePod()
+				ExpectProvisioned(ctx, env.Client, cluster, cloudProvider, coreProvisioner, pod)
+				ExpectScheduled(ctx, env.Client, pod)
```

Comment:
I also copy paste this bit a fair amount. Should we add a helper that encapsulates this? 

ExpectProvisionedNodeForPod()?

Reply:
which lines are you referring to?

Reply:
1018-1021